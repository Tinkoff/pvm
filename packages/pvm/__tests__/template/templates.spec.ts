import getEnv from '../../mechanics/template/env'

const template = `\
{%- macro showPkg(pkg) -%}
  *{{pkg.name}}@{{pkg.version}}*
{%- endmacro -%}
{{ success | items(0) | cutList("{restLen} more") | map(showPkg) | putCutMessage | joinAnd }}{{ " " }}
{{- success | enpl("all packages failed to release :(", "package has been released", "packages have been released") }}
{% for commit in commits | cutList("...") | putCutMessage -%}
  • {{ commit | stripServiceLabels | jiraLinkify }}
{% endfor -%}`

describe('templates', () => {
  it('шаблонизация списков через cut', async () => {
    const env = await getEnv()

    const ctx = {
      success: [
        [
          {
            name: 'test-a',
            version: '1.0.0',
          },
          '0.9.0',
        ],
        [
          {
            name: 'test-b',
            version: '2.0.0',
          },
          '1.3.0',
        ],
      ],
      commits: [
        'PFP-2321 Баг с получением полей структуры',
        'fix: коррекция наводки',
        'Добавлены валидаторы на максимальное кол-во символов для полей: `1`, `2`',
      ],
    }

    const result = env.renderString(template, ctx)

    expect(result).toMatchSnapshot()
  })

  it('шаблонизация списков через cut, 1 пакет', async () => {
    const env = await getEnv()

    const ctx = {
      success: [
        [
          {
            name: 'test-a',
            version: '1.0.0',
          },
          '0.9.0',
        ],
      ],
      commits: [
        'PFP-2321 Баг с получением полей структуры',
        'fix: коррекция наводки',
        'Добавлены валидаторы на максимальное кол-во символов для полей: `1`, `2`',
      ],
    }

    const result = env.renderString(template, ctx)

    expect(result).toMatchSnapshot()
  })
})
