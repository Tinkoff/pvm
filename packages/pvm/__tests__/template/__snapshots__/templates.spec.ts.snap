// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates шаблонизация списков через cut 1`] = `
"*test-a@1.0.0*, *test-b@2.0.0* packages have been released
• https://jira.example.com/browse/PFP-2321 Баг с получением полей структуры
• коррекция наводки
• Добавлены валидаторы на максимальное кол-во символов для полей: \`1\`, \`2\`
"
`;

exports[`templates шаблонизация списков через cut, 1 пакет 1`] = `
"*test-a@1.0.0* package has been released
• https://jira.example.com/browse/PFP-2321 Баг с получением полей структуры
• коррекция наводки
• Добавлены валидаторы на максимальное кол-во символов для полей: \`1\`, \`2\`
"
`;
